### PR TITLE
Added functionality for retrieving variables from control dependencies

### PR DESCRIPTION
**Pull request by Wapaul1**
This functionality is primarily for passing in `opt.minimize` as the loss function to `TensorFlowVariables`, and now no longer has the prefixing.

### CODE REVIEW

File: test/tensorflow_test.py [hunk 1]
```diff
@@ -187,5 +187,20 @@ def testNetworkDriverWorkerIndependent(self):
...
+    # Creates a network and appends a momentum optimizer.
+    sess = tf.Session()
+    loss, _ = make_linear_network()
+    minimizer = tf.train.MomentumOptimizer(0.9, 0.9).minimize(loss)
+    net_vars = ray.experimental.TensorFlowVariables(minimizer, sess)
+   
+    # Tests if all variables are properly retrieved, 2 variables and 2 momentum
+    # variables.
+    self.assertEqual(len(net_vars.variables.items()), 4)
+
```
Comment (robertnishihara) [inline test/tensorflow_test.py]: Can you also call `get_weights` and `set_weights` and make sure those methods work in this case? (you'll probably need to call init as well).

File: python/ray/experimental/tfutils.py [hunk 2]
```diff
@@ -42,10 +42,13 @@ def __init__(self, loss, sess=None, prefix=False):
     # We do a BFS on the dependency graph of the input function to find 
     # the variables.
     while len(queue) != 0:
-      op = queue.popleft().op
-      queue.extend(op.inputs)
-      if op.node_def.op == "Variable":
-        variable_names.append(op.node_def.name)
+      tf_obj = queue.popleft()
+      if hasattr(tf_obj, "op"):
```
Comment (robertnishihara) [inline python/ray/experimental/tfutils.py]: This if statement is a bit confusing to me, why are we doing this?

Comment (robertnishihara) [inline python/ray/experimental/tfutils.py]: If it's the right thing to do, could you document why?

**Comment by robertnishihara** [comment]
We also need to change `examples/lbfgs/driver.py` to use a separate graph for the network created there, right?

File: doc/using-ray-with-tensorflow.md [hunk 3]
```diff
@@ -95,11 +92,8 @@ NUM_BATCHES = 1
 NUM_ITERS = 201
 
 def net_vars_initializer():
-  # Prefix should be random so that there is no conflict with variable names in
-  # the cluster setting.
-  prefix = str(uuid.uuid1().hex)
-  # Use the tensorflow variable_scope to prefix all of the variables
-  with tf.variable_scope(prefix):
+  # Use a seperate graph for each network.
```
Comment (pcmoritz) [inline doc/using-ray-with-tensorflow.md]: seperate -> separate
